### PR TITLE
SG-168 Added User/Group mapping

### PR DESCRIPTION
**Pull request by azepam**
## Description
Added User/Group mapping

## Types of changes
- [x] New feature (non-breaking change which adds functionality)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fmc%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/admin-user-add.go [hunk 1]
```diff
@@ -166,7 +171,27 @@ func fetchUserKeys(args cli.Args) (string, string) {
 		secretKey = args.Get(2)
 	}
 
-	return accessKey, secretKey
+	if argCount < 4 {
```
Comment (yauhen-matsiushonak) [inline cmd/admin-user-add.go]: I would not require to enter these parameters if they are not provided as arguments and just populated them as nil.

File: cmd/admin-user-add.go [hunk 2]
```diff
@@ -133,16 +136,18 @@ func (u userMessage) JSON() string {
 }
 
 // fetchUserKeys - returns the access and secret key
```
Comment (yauhen-matsiushonak) [inline cmd/admin-user-add.go]: Should we rename the function then and change comment or should we put handling of additional arguments for mapped user/group into separate function?

**Review by yauhen-matsiushonak — COMMENTED** [review]
looks good as well, just some proposals added.

**Comment by yauhen-matsiushonak** [comment]
We also need to update help message for "user add" command. I guess this is missing just becase this is a draft change, so leave this as a reminder for further review.

File: cmd/admin-user-add.go (hunk 1 above)
Comment (azepam) [reply cmd/admin-user-add.go]: Done

File: cmd/admin-user-add.go (hunk 2 above)
Comment (azepam) [reply cmd/admin-user-add.go]: Renamed

File: cmd/admin-user-add.go [hunk 3]
```diff
@@ -54,17 +54,19 @@ FLAGS:
   {{range .VisibleFlags}}{{.}}
   {{end}}
 EXAMPLES:
-  1. Add a new user 'foobar' to MinIO server.
+  1. Add a new user 'foobar' with id '123' and group '456' to MinIO server.
```
Comment (yauhen-matsiushonak) [inline cmd/admin-user-add.go:57]: Should it be "uid:123" and "gid:456" instead of just numbers?

File: cmd/admin-user-add.go [hunk 4]
```diff
@@ -132,17 +137,19 @@ func (u userMessage) JSON() string {
 	return string(jsonMessageBytes)
 }
 
-// fetchUserKeys - returns the access and secret key
-func fetchUserKeys(args cli.Args) (string, string) {
+// fetchUserParams - returns the access and secret key, user and group id
+func fetchUserParams(args cli.Args) (string, string, string, string) {
 	accessKey := ""
 	secretKey := ""
+	sysUser := ""
```
Comment (yauhen-matsiushonak) [inline cmd/admin-user-add.go:144]: Should them be empty strings or nil? I'd like to not populate this data in database at all. Do we handle empty string and nil the same way?

Comment (azepam) [reply cmd/admin-user-add.go:144]: I put omitempty setting everywhere, so it shouldn't be in database

File: cmd/admin-user-add.go (hunk 3 above)
Comment (azepam) [reply cmd/admin-user-add.go:57]: Changed regexp in minio and help message in this repo

**Review by yauhen-matsiushonak — APPROVED** [review]

**Review by dmitrizpanasas — APPROVED** [review]

**Pull request merged**